docs/design/persistent-storage.md

```diff
@@ -0,0 +1,235 @@
+# Persistent Stores
+
+This document proposes a model for managing persistent, cluster-scoped storage for applications requiring long lived data.
+
+### tl;dr
+
+`PersistentStore` objects contain a VolumeSource with a lifecycle greater than the pods it is used by.  Many types of VolumeSources can be implemented as volume plugins, with each implementing an interface for use within a controller.  A pod with persistence requirements (specifically pod.spec.volumes.volume.persistenceRequirements) is bound to a matching VolumeSource from a pool of available stores.  PersistentStore objects give that VolumeSource longevity.
+
+A `PersistentStoreController` manages the creation, attachment, and reclamation of PersistentStore objects for one VolumeSource.  Implementations must be pluggable to support many types of controllers for many types of VolumeSources. The controller can advise the scheduler on host selection, keeping volume rules contained within the provider.  All cloud provider credentials are isolated to the store controller's host.  Every store in the cluster requires a controller to manage the pool (even a pool of 1).
```

Comment:
So a PSController is an API object that hosts plugins which implement the various kinds of volumes?  If it is an API object, what does "the store controller's host" mean?  An API object does not have a host.

You sort of introduce the word pool here, but I don't know what it means.  Are you saying that all PDs are controlled by a single PSController or that every PS has a PSController?